// react-cross-fader/dist/index.es.js
(function(){"use strict";try{if(typeof document<"u"){var e=document.createElement("style");e.appendChild(document.createTextNode(":root{--fade-in-animation: defaultFadeIn .2s ease;--fade-in-delay: 0s;--fade-out-animation: defaultFadeOut .2s ease;--fade-out-delay: 0s}@keyframes defaultFadeIn{0%{opacity:0}to{opacity:1}}@keyframes defaultFadeOut{to{opacity:0}}.react-cross-fader-container .react-cross-fader-element{width:100%;height:100%;position:absolute}.react-cross-fader-container .react-cross-fader-element.fade-in{opacity:0;animation:var(--fade-in-animation) forwards;animation-delay:var(--fade-in-delay);z-index:20}.react-cross-fader-container .react-cross-fader-element.fade-out{pointer-events:none;animation:var(--fade-out-animation) forwards;animation-delay:var(--fade-out-delay);z-index:19}")),document.head.appendChild(e)}}catch(a){console.error("vite-plugin-css-injected-by-js",a)}})();
import Le, { forwardRef as fr, useRef as Fe, useState as j, useEffect as lr } from "react";
var ce = { exports: {} }, Y = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var $e;
function cr() {
  if ($e)
    return Y;
  $e = 1;
  var V = Le, _ = Symbol.for("react.element"), $ = Symbol.for("react.fragment"), g = Object.prototype.hasOwnProperty, P = V.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, C = { key: !0, ref: !0, __self: !0, __source: !0 };
  function y(R, d, O) {
    var f, v = {}, T = null, k = null;
    O !== void 0 && (T = "" + O), d.key !== void 0 && (T = "" + d.key), d.ref !== void 0 && (k = d.ref);
    for (f in d)
      g.call(d, f) && !C.hasOwnProperty(f) && (v[f] = d[f]);
    if (R && R.defaultProps)
      for (f in d = R.defaultProps, d)
        v[f] === void 0 && (v[f] = d[f]);
    return { $$typeof: _, type: R, key: T, ref: k, props: v, _owner: P.current };
  }
  return Y.Fragment = $, Y.jsx = y, Y.jsxs = y, Y;
}
var N = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ie;
function dr() {
  return Ie || (Ie = 1, process.env.NODE_ENV !== "production" && function() {
    var V = Le, _ = Symbol.for("react.element"), $ = Symbol.for("react.portal"), g = Symbol.for("react.fragment"), P = Symbol.for("react.strict_mode"), C = Symbol.for("react.profiler"), y = Symbol.for("react.provider"), R = Symbol.for("react.context"), d = Symbol.for("react.forward_ref"), O = Symbol.for("react.suspense"), f = Symbol.for("react.suspense_list"), v = Symbol.for("react.memo"), T = Symbol.for("react.lazy"), k = Symbol.for("react.offscreen"), M = Symbol.iterator, Q = "@@iterator";
    function I(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = M && e[M] || e[Q];
      return typeof r == "function" ? r : null;
    }
    var S = V.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function c(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        ee("error", e, t);
      }
    }
    function ee(e, r, t) {
      {
        var n = S.ReactDebugCurrentFrame, o = n.getStackAddendum();
        o !== "" && (r += "%s", t = t.concat([o]));
        var u = t.map(function(i) {
          return String(i);
        });
        u.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, u);
      }
    }
    var h = !1, re = !1, U = !1, B = !1, q = !1, J;
    J = Symbol.for("react.module.reference");
    function G(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === g || e === C || q || e === P || e === O || e === f || B || e === k || h || re || U || typeof e == "object" && e !== null && (e.$$typeof === T || e.$$typeof === v || e.$$typeof === y || e.$$typeof === R || e.$$typeof === d || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === J || e.getModuleId !== void 0));
    }
    function L(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var o = r.displayName || r.name || "";
      return o !== "" ? t + "(" + o + ")" : t;
    }
    function A(e) {
      return e.displayName || "Context";
    }
    function E(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && c("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case g:
          return "Fragment";
        case $:
          return "Portal";
        case C:
          return "Profiler";
        case P:
          return "StrictMode";
        case O:
          return "Suspense";
        case f:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case R:
            var r = e;
            return A(r) + ".Consumer";
          case y:
            var t = e;
            return A(t._context) + ".Provider";
          case d:
            return L(e, e.render, "ForwardRef");
          case v:
            var n = e.displayName || null;
            return n !== null ? n : E(e.type) || "Memo";
          case T: {
            var o = e, u = o._payload, i = o._init;
            try {
              return E(i(u));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var b = Object.assign, x = 0, de, ve, pe, Ee, me, be, ge;
    function Re() {
    }
    Re.__reactDisabledLog = !0;
    function We() {
      {
        if (x === 0) {
          de = console.log, ve = console.info, pe = console.warn, Ee = console.error, me = console.group, be = console.groupCollapsed, ge = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: Re,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        x++;
      }
    }
    function Ye() {
      {
        if (x--, x === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: b({}, e, {
              value: de
            }),
            info: b({}, e, {
              value: ve
            }),
            warn: b({}, e, {
              value: pe
            }),
            error: b({}, e, {
              value: Ee
            }),
            group: b({}, e, {
              value: me
            }),
            groupCollapsed: b({}, e, {
              value: be
            }),
            groupEnd: b({}, e, {
              value: ge
            })
          });
        }
        x < 0 && c("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var te = S.ReactCurrentDispatcher, ne;
    function z(e, r, t) {
      {
        if (ne === void 0)
          try {
            throw Error();
          } catch (o) {
            var n = o.stack.trim().match(/\n( *(at )?)/);
            ne = n && n[1] || "";
          }
        return `
` + ne + e;
      }
    }
    var ae = !1, H;
    {
      var Ne = typeof WeakMap == "function" ? WeakMap : Map;
      H = new Ne();
    }
    function he(e, r) {
      if (!e || ae)
        return "";
      {
        var t = H.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      ae = !0;
      var o = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var u;
      u = te.current, te.current = null, We();
      try {
        if (r) {
          var i = function() {
            throw Error();
          };
          if (Object.defineProperty(i.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(i, []);
            } catch (w) {
              n = w;
            }
            Reflect.construct(e, [], i);
          } else {
            try {
              i.call();
            } catch (w) {
              n = w;
            }
            e.call(i.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (w) {
            n = w;
          }
          e();
        }
      } catch (w) {
        if (w && n && typeof w.stack == "string") {
          for (var a = w.stack.split(`
`), p = n.stack.split(`
`), s = a.length - 1, l = p.length - 1; s >= 1 && l >= 0 && a[s] !== p[l]; )
            l--;
          for (; s >= 1 && l >= 0; s--, l--)
            if (a[s] !== p[l]) {
              if (s !== 1 || l !== 1)
                do
                  if (s--, l--, l < 0 || a[s] !== p[l]) {
                    var m = `
` + a[s].replace(" at new ", " at ");
                    return e.displayName && m.includes("<anonymous>") && (m = m.replace("<anonymous>", e.displayName)), typeof e == "function" && H.set(e, m), m;
                  }
                while (s >= 1 && l >= 0);
              break;
            }
        }
      } finally {
        ae = !1, te.current = u, Ye(), Error.prepareStackTrace = o;
      }
      var F = e ? e.displayName || e.name : "", De = F ? z(F) : "";
      return typeof e == "function" && H.set(e, De), De;
    }
    function Ve(e, r, t) {
      return he(e, !1);
    }
    function Me(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function X(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return he(e, Me(e));
      if (typeof e == "string")
        return z(e);
      switch (e) {
        case O:
          return z("Suspense");
        case f:
          return z("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            return Ve(e.render);
          case v:
            return X(e.type, r, t);
          case T: {
            var n = e, o = n._payload, u = n._init;
            try {
              return X(u(o), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var K = Object.prototype.hasOwnProperty, _e = {}, ye = S.ReactDebugCurrentFrame;
    function Z(e) {
      if (e) {
        var r = e._owner, t = X(e.type, e._source, r ? r.type : null);
        ye.setExtraStackFrame(t);
      } else
        ye.setExtraStackFrame(null);
    }
    function Ue(e, r, t, n, o) {
      {
        var u = Function.call.bind(K);
        for (var i in e)
          if (u(e, i)) {
            var a = void 0;
            try {
              if (typeof e[i] != "function") {
                var p = Error((n || "React class") + ": " + t + " type `" + i + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[i] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw p.name = "Invariant Violation", p;
              }
              a = e[i](r, i, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (s) {
              a = s;
            }
            a && !(a instanceof Error) && (Z(o), c("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, i, typeof a), Z(null)), a instanceof Error && !(a.message in _e) && (_e[a.message] = !0, Z(o), c("Failed %s type: %s", t, a.message), Z(null));
          }
      }
    }
    var Be = Array.isArray;
    function ie(e) {
      return Be(e);
    }
    function qe(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Je(e) {
      try {
        return Te(e), !1;
      } catch {
        return !0;
      }
    }
    function Te(e) {
      return "" + e;
    }
    function we(e) {
      if (Je(e))
        return c("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", qe(e)), Te(e);
    }
    var W = S.ReactCurrentOwner, Ge = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Pe, Oe, oe;
    oe = {};
    function ze(e) {
      if (K.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function He(e) {
      if (K.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Xe(e, r) {
      if (typeof e.ref == "string" && W.current && r && W.current.stateNode !== r) {
        var t = E(W.current.type);
        oe[t] || (c('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', E(W.current.type), e.ref), oe[t] = !0);
      }
    }
    function Ke(e, r) {
      {
        var t = function() {
          Pe || (Pe = !0, c("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function Ze(e, r) {
      {
        var t = function() {
          Oe || (Oe = !0, c("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var Qe = function(e, r, t, n, o, u, i) {
      var a = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: _,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: i,
        // Record the component responsible for creating this element.
        _owner: u
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: o
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function er(e, r, t, n, o) {
      {
        var u, i = {}, a = null, p = null;
        t !== void 0 && (we(t), a = "" + t), He(r) && (we(r.key), a = "" + r.key), ze(r) && (p = r.ref, Xe(r, o));
        for (u in r)
          K.call(r, u) && !Ge.hasOwnProperty(u) && (i[u] = r[u]);
        if (e && e.defaultProps) {
          var s = e.defaultProps;
          for (u in s)
            i[u] === void 0 && (i[u] = s[u]);
        }
        if (a || p) {
          var l = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && Ke(i, l), p && Ze(i, l);
        }
        return Qe(e, a, p, o, n, W.current, i);
      }
    }
    var ue = S.ReactCurrentOwner, Se = S.ReactDebugCurrentFrame;
    function D(e) {
      if (e) {
        var r = e._owner, t = X(e.type, e._source, r ? r.type : null);
        Se.setExtraStackFrame(t);
      } else
        Se.setExtraStackFrame(null);
    }
    var se;
    se = !1;
    function fe(e) {
      return typeof e == "object" && e !== null && e.$$typeof === _;
    }
    function Ce() {
      {
        if (ue.current) {
          var e = E(ue.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function rr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var xe = {};
    function tr(e) {
      {
        var r = Ce();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function je(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = tr(r);
        if (xe[t])
          return;
        xe[t] = !0;
        var n = "";
        e && e._owner && e._owner !== ue.current && (n = " It was passed a child from " + E(e._owner.type) + "."), D(e), c('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), D(null);
      }
    }
    function ke(e, r) {
      {
        if (typeof e != "object")
          return;
        if (ie(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            fe(n) && je(n, r);
          }
        else if (fe(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var o = I(e);
          if (typeof o == "function" && o !== e.entries)
            for (var u = o.call(e), i; !(i = u.next()).done; )
              fe(i.value) && je(i.value, r);
        }
      }
    }
    function nr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === d || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === v))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = E(r);
          Ue(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !se) {
          se = !0;
          var o = E(r);
          c("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", o || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && c("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ar(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            D(e), c("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), D(null);
            break;
          }
        }
        e.ref !== null && (D(e), c("Invalid attribute `ref` supplied to `React.Fragment`."), D(null));
      }
    }
    function Ae(e, r, t, n, o, u) {
      {
        var i = G(e);
        if (!i) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var p = rr(o);
          p ? a += p : a += Ce();
          var s;
          e === null ? s = "null" : ie(e) ? s = "array" : e !== void 0 && e.$$typeof === _ ? (s = "<" + (E(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : s = typeof e, c("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", s, a);
        }
        var l = er(e, r, t, o, u);
        if (l == null)
          return l;
        if (i) {
          var m = r.children;
          if (m !== void 0)
            if (n)
              if (ie(m)) {
                for (var F = 0; F < m.length; F++)
                  ke(m[F], e);
                Object.freeze && Object.freeze(m);
              } else
                c("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              ke(m, e);
        }
        return e === g ? ar(l) : nr(l), l;
      }
    }
    function ir(e, r, t) {
      return Ae(e, r, t, !0);
    }
    function or(e, r, t) {
      return Ae(e, r, t, !1);
    }
    var ur = or, sr = ir;
    N.Fragment = g, N.jsx = ur, N.jsxs = sr;
  }()), N;
}
process.env.NODE_ENV === "production" ? ce.exports = cr() : ce.exports = dr();
var le = ce.exports;
const pr = fr(
  function({
    children: _,
    className: $,
    onFadeOutComplete: g,
    onFadeInComplete: P,
    destroyOnFadeOutComplete: C = !0,
    distinctKey: y,
    delay: R = 0,
    ...d
  }, O) {
    const f = Fe(null), v = Fe(null), [T, k] = j(!1), [M, Q] = j(null), [I, S] = j(null), [c, ee] = j(null), [h, re] = j(0), [U, B] = j(!1), [q, J] = j(null);
    lr(() => {
      R ? (q && clearTimeout(q), J(setTimeout(() => {
        G(_, y);
      }, R))) : G(_, y);
    }, [_, y]);
    const G = (E, b) => {
      E === c || b && b === M || (B(!1), ee(I), Q(b ?? null), S(E), re((x) => (x + 1) % 2));
    }, L = () => {
      B(!0), g == null || g();
    }, A = () => {
      P == null || P();
    };
    return T || (f.current && (f.current.addEventListener("webkitAnimationEnd", L, !1), f.current.addEventListener("animationend", L, !1), f.current.addEventListener("onanimationend", L, !1)), v.current && (v.current.addEventListener("webkitAnimationEnd", A, !1), v.current.addEventListener("animationend", A, !1), v.current.addEventListener("onanimationend", A, !1)), k(!0)), /* @__PURE__ */ le.jsxs("div", { ...d, ref: O, className: `react-cross-fader-container ${$ ?? ""}`, children: [
      /* @__PURE__ */ le.jsx(
        "div",
        {
          ref: h === 0 ? v : f,
          className: `react-cross-fader-element ${h === 0 ? "fade-in" : "fade-out"}`,
          children: !(U && h === 1 && C) && (h === 0 ? I : c)
        }
      ),
      /* @__PURE__ */ le.jsx(
        "div",
        {
          ref: h === 1 ? v : f,
          className: `react-cross-fader-element ${h === 1 ? "fade-in" : "fade-out"}`,
          children: !(U && h === 0 && C) && (h === 1 ? I : c)
        }
      )
    ] });
  }
);
export {
  pr as CrossFader
};
